contracts/FlyoverDiscovery.sol

```diff
@@ -0,0 +1,266 @@
+        return _getProvider(providerAddress);
+    }
+
+    /// @notice Checks if an LP can operate for peg-in side
+    /// @dev Ignores the first argument as compatibility stub with legacy signature
+    /// @param addr The LP address
+    /// @return isOp True if registered and peg-in collateral >= min
+    function isOperational(Flyover.ProviderType, address addr) external view returns (bool) {
+        return _isRegistered(addr) && _collateralManagement.getPegInCollateral(addr) >= _minCollateral;
+    }
+
+    /// @notice Checks if an LP can operate for peg-out side
+    /// @param addr The LP address
+    /// @return isOp True if registered-for-pegout and peg-out collateral >= min
+    function isOperationalForPegout(address addr) external view returns (bool) {
+        return _isRegisteredForPegout(addr) && _collateralManagement.getPegOutCollateral(addr) >= _minCollateral;
+    }
+
+    // ------------------------------------------------------------
+    // Getter Functions
+    // ------------------------------------------------------------
+
+    /// @notice Returns the current minimum collateral threshold per side
+    /// @return minCollateral The minimum collateral value
+    function getMinCollateral() external view returns (uint256) {
+        return _minCollateral;
+    }
+
+    /// @notice Returns the last assigned provider id
+    /// @return lastId Last provider id
+    function getProvidersId() external view returns (uint) {
+        return lastProviderId;
+    }
+
+    /// @notice Returns the resignation starting block for an LP
+    /// @param addr The LP address
+    /// @return resignationBlock Block number when resignation started, or 0 if not resigned
+    function getResignationBlockNum(address addr) external view returns (uint256) {
+        return _resignationBlockNum[addr];
+    }
+
+    // ------------------------------------------------------------
+    // FlyoverDiscovery Private Functions
+    // ------------------------------------------------------------
+
+    function _shouldBeListed(Flyover.LiquidityProvider storage lp) private view returns(bool){
+        return (_isRegistered(lp.providerAddress) || _isRegisteredForPegout(lp.providerAddress)) && lp.status;
+    }
+
+    function _validateRegistration(
+        string memory name,
+        string memory apiBaseUrl,
+        Flyover.ProviderType providerType,
+        address providerAddress
+    ) private view {
+        if (providerAddress != msg.sender || providerAddress.code.length != 0) revert NotEOA(providerAddress);
+
+        if (
+            bytes(name).length < 1 ||
+            bytes(apiBaseUrl).length < 1
+        ) {
+            revert InvalidProviderData(name, apiBaseUrl);
+        }
+
+        if (providerType > type(Flyover.ProviderType).max) revert InvalidProviderType(providerType);
+
+        if (_resignationBlockNum[providerAddress] != 0) {
```

Comment:
this condition and the ones below should change since the minCollateral doesn't go on discovery, you can check here for reference https://github.com/rsksmart/liquidity-bridge-contract/blob/lbc-split/contracts/split/FlyoverDiscovery.sol#L138